client/components/data/query-stats-posts-views/index.jsx

```diff
@@ -0,0 +1,57 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import shallowEqual from 'react-pure-render/shallowEqual';
+
+/**
+ * Internal dependencies
+ */
+import { requestPostsViews } from 'state/stats/views/posts/actions';
+
+class QueryPostsViews extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		postIds: PropTypes.arrayOf( PropTypes.number ),
+		num: PropTypes.number,
+		date: PropTypes.string,
+	};
+
+	componentDidMount() {
+		this.request();
+	}
+
+	componentDidUpdate( prevProps ) {
```

Comment:
Do we need to check `num` and `date` props here as well, in case those have changed when a re-render is triggered?